src/components/spreadsheet/table-wrapper.js

```diff
@@ -0,0 +1,686 @@
+
+            if (column.numeric) {
+                column.cellRendererParams = {
+                    loadFlowStatus: props.loadFlowStatus,
+                    network: props.network,
+                };
+
+                if (column.normed) {
+                    column.cellRendererParams.fluxConvention = fluxConvention;
+                }
+            }
+
+            column.width = column.columnWidth || MIN_COLUMN_WIDTH;
+            column.pinned = lockedColumnsNames.has(column.id)
+                ? 'left'
+                : undefined;
+
+            return column;
+        },
+        [
+            fluxConvention,
+            intl,
+            lockedColumnsNames,
+            props.loadFlowStatus,
+            props.network,
+        ]
+    );
+
+    const addEditColumn = useCallback(
+        (columns) => {
+            columns.unshift({
+                field: 'edit',
+                locked: true,
+                pinned: 'left',
+                lockPosition: 'left',
+                sortable: false,
+                filter: false,
+                resizable: false,
+                width: 100,
+                headerName: '',
+                cellRendererSelector: (params) => {
+                    if (params.node.rowPinned) {
+                        return {
+                            component: EditingCellRenderer,
+                            params: {
+                                setEditingData: setEditingData,
+                                setIsValidatingData: setIsValidatingData,
+                                startEditing: startEditing,
+                            },
+                        };
+                    } else if (editingData?.id === params.data.id) {
+                        return {
+                            component: ReferenceLineCellRenderer,
+                        };
+                    } else {
+                        return {
+                            component: EditableCellRenderer,
+                            params: {
+                                setEditingData: setEditingData,
+                                equipmentType:
+                                    TABLES_DEFINITION_INDEXES.get(tabIndex)
+                                        .modifiableEquipmentType,
+                            },
+                        };
+                    }
+                },
+            });
+        },
+        [editingData?.id, startEditing, tabIndex]
+    );
+
+    const generateTableColumns = useCallback(
+        (tabIndex) => {
+            const generatedTableColumns = TABLES_DEFINITION_INDEXES.get(
+                tabIndex
+            )
+                .columns.filter((c) => {
+                    return selectedColumnsNames.has(c.id);
+                })
+                .map((column) => enrichColumn(column));
+
+            function sortByIndex(a, b) {
+                if (reorderedTableDefinitionIndexes) {
+                    if (
```

Comment:
Why not just
reorderedTableDefinitionIndexes.indexOf(a.id) - reorderedTableDefinitionIndexes.indexOf(b.id)
?

Reply:
Seems to do the trick and is more concise, I implemented it